### DISCUSSION TITLE
`restrictTemplateExpressions`, `@typescript-eslint/restrict-template-expressions`

### DISCUSSION THREAD

**Discussion by jpike88** [rome/tools#4347] [Suggestions]
https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/restrict-template-expressions.md

JavaScript will call toString() on an object when it is converted to a string, such as when + adding to a string or in ${} template literals. The default Object .toString() returns "[object Object]", which is often not what was intended. This rule reports on values used in a template literal string that aren't primitives and don't define a more useful .toString() method.

**Comment by Conaclos** [comment]
This rule requires type info.
